### ISSUE TITLE
400MB limit using s3 links and resulting: decompression failed

### ISSUE THREAD

**Issue by pjammer** [fringd/zipline#38]
Hi,

I have a list of say 43 s3 links, each with a varying filesize all links are to `MOV` files. So some are big, some a little clips.

while i can get a zip file created with 40 small clips, for some reason i'm hitting a wall with 400MB. Changing my nginx settings to `proxy_buffering no` hasn't helped. i get this issue with `curl -LO url/zip` and through the browser.  The code that i am using:

```ruby
  def zip

    @video_clip_info = @current_account.videos.find(params[:id])
    @video_clips = if params[:after].present?
                     @video_clip_info.video_clips.clips_to_download(params["after"].values.join('-').to_date).order('created_at DESC')
                   else
                     @video_clip_info.video_clips.order('created_at DESC')
                   end

# there are 43 videos verified with puts statements and logging.
    @files =  if @video_clips.present?
# passing the vid.clip (Carrierwave object) and i chunk out the file name 
                @video_clips.map{ |vid| [vid.clip, "#{vid.clip.filename.split(/\//).last}"] }
              else
                []
              end
# didn't have time to learn how to stub out properly so this is what i do in test env.
    unless Rails.env.test? || @files.blank?
      zipline( @files, "video_#{@video_clip_info.id}_#{Time.now.to_i}.zip")
    else
      respond_to do |format|
        format.html {render text: "hello world"}
      end
    end
  end
```

I can individually download each file with no problem as some are greater than 400MB and they download, so i am not sure where to go or what to do at this point.   

If it's an s3 problem, why can i download ONE file; 
if it's a Zipline issue (or one that i am coding) why doesn't `proxy_buffering` not help?  Am i missing a setting in Zipline that i can set to allow a very large Zip to be created.

Thoughts and/or help would be fantastic. It's been a while of bashing head against wall for me so need some new eyes on this.

**Comment by fringd** [comment]
It might be a limitation of the servers you're running on. Are you using
heroku or something else? Are there any exceptions in the server logs?

**Comment by fringd** [comment]
Downloading directly is probably going straight to s3

**Comment by pjammer** [comment]
yeah i hope the downloading is going through s3; server is on Opsworks (AWS) and no real error messages on the servers as that `zip` code 200s and _maybe_ that is it for my server. the point was to deal with s3 and the browser once my code gave it all the urls. 

It's still a bit magic for me though, Zipline. I'll search for more logs but nothing in `production.log`

**Comment by fringd** [comment]
or maybe this one is helpful: https://stackoverflow.com/questions/13312132/nginx-unicorn-do-not-allow-upload-file-400mb/39676919

**Comment by pjammer** [comment]
sweet. will read up and update. Thanks man.

**Comment by pjammer** [comment]
i lied... though i would have sworn it was puma, due to changes in the deploy recipe.... it is unicorn that we are using. Does unicorn support streaming properly, do you know?

**Comment by fringd** [comment]
unicorn... I'm not sure if that one works out of the box.

**Comment by fringd** [comment]
you have to configure it, dunno if you already did that:
https://api.rubyonrails.org/classes/ActionController/Streaming.html#module-ActionController::Streaming-label-Unicorn

**Comment by fringd** [comment]
no problem man, glad you got it working ^_^
